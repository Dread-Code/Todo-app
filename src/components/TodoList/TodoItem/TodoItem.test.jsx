import React from "react";
import { shallow } from "enzyme";
import { TodoItem } from "./TodoItem";

describe("TodoItem", () => {
  const handleToggle = jest.fn();
  const deleteTodo = jest.fn();
  const todo = {
    id: 1625964511944,
    desc: "Estudiar jest",
    done: false,
  };
  const wrapper = shallow(
    <TodoItem
      todo={todo}
      index={0}
      handleToggle={handleToggle}
      deleteTodo={deleteTodo}
    />
  );

  test("should match to snapshoot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should call the handleToggle function", () => {
    wrapper.find("p").simulate("click");
    expect(handleToggle).toHaveBeenCalledWith(todo.id);
  });

  test("should call the deleteTodo function", () => {
    wrapper.find("button").simulate("click");
    expect(deleteTodo).toHaveBeenCalledWith(todo.id);
  });

  test("should show the text correctlty", () => {
    const p = wrapper.find("p");
    expect(p.text()).toBe(`1. ${todo.desc}`);
  });

  test("should have the complete class if todo.done equal true", () => {
    const newTodo = {
      ...todo,
      done: !todo.done,
    };

    const wrapper = shallow(
      <TodoItem
        todo={newTodo}
        index={0}
        handleToggle={handleToggle}
        deleteTodo={deleteTodo}
      />
    );
    const p = wrapper.find("p");

    expect(wrapper).toMatchSnapshot();
    expect(p.hasClass("complete")).toBe(true);
  });
});
