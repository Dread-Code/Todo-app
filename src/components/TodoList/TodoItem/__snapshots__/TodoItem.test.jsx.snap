// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem should have the complete class if todo.done equal true 1`] = `
<Fragment>
  <li
    className="list-group-item"
    key="1625964511944"
  >
    <p
      className="complete"
      onClick={[Function]}
    >
      1
      . 
      Estudiar jest
    </p>
    <button
      className="btn btn-danger"
      onClick={[Function]}
    >
      Eliminar
    </button>
  </li>
</Fragment>
`;

exports[`TodoItem should match to snapshoot 1`] = `
<Fragment>
  <li
    className="list-group-item"
    key="1625964511944"
  >
    <p
      className="false"
      onClick={[Function]}
    >
      1
      . 
      Estudiar jest
    </p>
    <button
      className="btn btn-danger"
      onClick={[Function]}
    >
      Eliminar
    </button>
  </li>
</Fragment>
`;
